### ISSUE TITLE
Slug routes not building in development

### ISSUE THREAD

**Issue by aewing** [Elderjs/elderjs#25]
In development, running `npm run dev:rollup` and `npm run dev:server` as instructed, routes in `blog/route.js` are not being built.

For example, with the following in `src/routes/blog/route.js`, `/blog/hello-world` is "Not Found":
```js
module.exports = {
  all: () => [{ slug: 'hello-world' }],
  permalink: ({ request }) => `/blog/${request.slug}`,
};
```

After building with `npm run build` and starting the dev server again, the route loads.
Subsequent changes to the Blog template do not cause `dev:rollup` builds and are not reflected until `npm run build` is executed again.
Is this expected behavior?

**Comment by nickreese** [comment]
@aewing 

Try opening two terminals and running `npm run dev:rollup` and `npm run dev:server` I'm going to be pushing some changes to the template to make the dev process smoother shortly. 

Let me know if that doesn't work for you.

**Comment by aewing** [comment]
@nickreese I am running both terminal processes. I'm not receiving any errors in either terminal process, but `/blog/hello-world` is "Not Found" until I run `npm run build`.

**Comment by aewing** [comment]
A reproduction based on the current template: https://github.com/aewing/elderjs-bug-repro

**Comment by aewing** [comment]
Another error I noticed with the reproduction, is that the lack of a component in `components/` causes a rollup error when running `npm run build:rollup`. This is a separate issue.

**Comment by aewing** [comment]
^ Expanding on the above, I had to have a component inside of a subdirectory of `src/components` to resolve this error.

**Comment by nickreese** [comment]
What is your dev env? 

Agreed the rollup bug is a separate issue. 

This appears to be an issue with nodemon not detecting the changes. Will try and recreate with the repo you linked.

**Comment by nickreese** [comment]
OK sorted it out. It appears that we're assuming a trailing slash in Elder.js. Add in a trailing slash in your `route.js` and it works.

```
module.exports = {
  all: () => [{ slug: 'hello-world' }, { slug: 'foo-bar' }],
  permalink: ({ request }) => `/blog/${request.slug}/`,
};
```

**Comment by nickreese** [comment]
That said, you've also deleted the plugin in your test project, so there is no data to read.

**Comment by nickreese** [comment]
I've fixed the rollup on the template as well. It won't try and rollup an empty components folder.

**Comment by aewing** [comment]
Thanks for looking into this. Is the need for a trailing slash something you expect will change in the near future? If not, if you'd like, I can contribute some language to the documentation to make this requirement clearer.

In regards to the plugin, I was just trying to demonstrate the issue I was seeing when developing with slugs coming from an external source with the smallest amount of code possible.

**Comment by aewing** [comment]
I would also be willing to make an attempt at a fix if you point me in the right direction.

**Comment by nickreese** [comment]
@aewing Got a fix coming shortly. :) 

Basically Elder.js was ported from how we used it to build ElderGuide.com so some assumptions (like trailing slash) are still lingering. This was one of them. 

We'll now check to see if we have a route as the request comes in. /blog/foo if we don't find it, we'll check /blog/foo/. Same with /blog/foo/, we'll check /blog/foo.

**Comment by nickreese** [comment]
@aewing fixed in 0.1.3 and updated in the template.

**Comment by nickreese** [comment]
@aewing thanks for the help identifying these bugs. :)
